[1m============================= test session starts ==============================[0m
platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0 -- /usr/bin/python3
cachedir: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache
rootdir: /home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing
plugins: launch-testing-1.0.10, ament-lint-0.12.14, ament-copyright-0.12.14, ament-xmllint-0.12.14, launch-testing-ros-0.19.10, ament-pep257-0.12.14, ament-flake8-0.12.14, colcon-core-0.19.0, cov-3.0.0
[1mcollecting ... [0m[1m
collected 1 item                                                               [0m

test/test_processor_service.py::test_processor_service [DEBUG] [launch.launch_context]: emitting event synchronously: 'launch.events.IncludeLaunchDescription'
[INFO] [launch]: All log files can be found below /home/phr/.ros/log/2025-08-07-12-16-33-626592-PHR-13712
[INFO] [launch]: Default logging verbosity is set to DEBUG
[DEBUG] [launch]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x720cdea79e70>'
[DEBUG] [launch]: processing event: '<launch.events.include_launch_description.IncludeLaunchDescription object at 0x720cdea79e70>' âœ“ '<launch.event_handlers.on_include_launch_description.OnIncludeLaunchDescription object at 0x720cddd822c0>'
[INFO] [number_processor-1]: process started with pid [13729]
[DEBUG] [launch.launch_context]: emitting event: 'launch.events.process.ProcessStarted'
[DEBUG] [launch]: processing event: '<launch.events.process.process_started.ProcessStarted object at 0x720cdd3d0400>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_started.ProcessStarted object at 0x720cdd3d0400>' âœ“ '<launch.event_handlers.on_process_start.OnProcessStart object at 0x720cdea7ba60>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_started.ProcessStarted object at 0x720cdd3d0400>' âœ“ '<launch.event_handlers.on_process_start.OnProcessStart object at 0x720cdea78430>'
[DEBUG] [launch.launch_context]: emitting event synchronously: 'launch.events.process.ProcessStderr'
[DEBUG] [launch]: processing event: '<launch.events.process.process_stderr.ProcessStderr object at 0x720cdd3d05b0>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_stderr.ProcessStderr object at 0x720cdd3d05b0>' âœ“ '<launch.event_handlers.on_process_io.OnProcessIO object at 0x720cdeabbe80>'
[number_processor-1] Traceback (most recent call last):
[number_processor-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 33, in <module>
[number_processor-1]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_processor')())
[number_processor-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 25, in importlib_load_entry_point
[number_processor-1]     return next(matches).load()
[number_processor-1]   File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 171, in load
[DEBUG] [launch]: processing event: '<launch.events.process.process_stderr.ProcessStderr object at 0x720cdd3d05b0>' âœ“ '<launch.event_handlers.on_process_io.OnProcessIO object at 0x720cdea7b5b0>'
[DEBUG] [launch.launch_context]: emitting event synchronously: 'launch.events.process.ProcessStderr'
[DEBUG] [launch]: processing event: '<launch.events.process.process_stderr.ProcessStderr object at 0x720cdd3d0490>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_stderr.ProcessStderr object at 0x720cdd3d0490>' âœ“ '<launch.event_handlers.on_process_io.OnProcessIO object at 0x720cdeabbe80>'
[number_processor-1]     module = import_module(match.group('module'))
[number_processor-1]   File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
[number_processor-1]     return _bootstrap._gcd_import(name[level:], package, level)
[number_processor-1]   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
[number_processor-1]   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
[number_processor-1]   File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
[number_processor-1] ModuleNotFoundError: No module named 'my_algorithms.number_processor_service_node'
[DEBUG] [launch]: processing event: '<launch.events.process.process_stderr.ProcessStderr object at 0x720cdd3d0490>' âœ“ '<launch.event_handlers.on_process_io.OnProcessIO object at 0x720cdea7b5b0>'
[ERROR] [number_processor-1]: process has died [pid 13729, exit code 1, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor --ros-args -r __node:=number_processor_service_node'].
[DEBUG] [launch.launch_context]: emitting event: 'launch.events.process.ProcessExited'
[DEBUG] [launch]: processing event: '<launch.events.process.process_exited.ProcessExited object at 0x720cdd3d00a0>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_exited.ProcessExited object at 0x720cdd3d00a0>' âœ“ '<launch.event_handlers.on_process_exit.OnProcessExit object at 0x720cdeabbee0>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_exited.ProcessExited object at 0x720cdd3d00a0>' âœ“ '<launch.event_handlers.on_process_exit.OnProcessExit object at 0x720cdeab9c60>'
[DEBUG] [launch]: processing event: '<launch.events.process.process_exited.ProcessExited object at 0x720cdd3d00a0>' âœ“ '<launch.event_handlers.on_process_exit.OnProcessExit object at 0x720cdea78340>'
[DEBUG] [launch.launch_context]: emitting event: 'launch.events.Shutdown'
[DEBUG] [launch]: processing event: '<launch.events.shutdown.Shutdown object at 0x720cddd83d60>'
[DEBUG] [launch]: processing event: '<launch.events.shutdown.Shutdown object at 0x720cddd83d60>' âœ“ '<launch.event_handlers.on_shutdown.OnShutdown object at 0x720cdeab9d20>'
[DEBUG] [launch]: processing event: '<launch.events.shutdown.Shutdown object at 0x720cddd83d60>' âœ“ '<launch.event_handlers.on_shutdown.OnShutdown object at 0x720cddd823e0>'
Processes under test stopped before tests completed
Process 'number_processor-1' exited with 1
##### 'number_processor-1' output #####
Traceback (most recent call last):
  File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 33, in <module>
    sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_processor')())
  File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 171, in load

    module = import_module(match.group('module'))
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'my_algorithms.number_processor_service_node'

#######################################
[31mFAILED[0m

=================================== FAILURES ===================================
[31m[1m_____________________ launch tests: test_processor_service _____________________[0m

===========================================================
FAIL: test_processor_service.TestProcessorService.test_service
-----------------------------------------------------------
Exception: Launch stopped before the active tests finished.
------------------------------ Captured log call -------------------------------
[35mDEBUG   [0m asyncio:selector_events.py:54 Using selector: EpollSelector
[35mDEBUG   [0m asyncio:base_events.py:1616 execute program '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor' stdin=<pipe> stdout=<pipe> stderr=<pipe>
[35mDEBUG   [0m asyncio:base_subprocess.py:50 process '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor' created: pid 13729
[35mDEBUG   [0m asyncio:base_events.py:1601 Write pipe 27 connected: (<_UnixWritePipeTransport fd=27 idle bufsize=0>, <WriteSubprocessPipeProto fd=0 pipe=<_UnixWritePipeTransport fd=27 idle bufsize=0>>)
[35mDEBUG   [0m asyncio:base_events.py:1585 Read pipe 28 connected: (<_UnixReadPipeTransport fd=28 polling>, <ReadSubprocessPipeProto fd=1 pipe=<_UnixReadPipeTransport fd=28 polling>>)
[35mDEBUG   [0m asyncio:base_events.py:1585 Read pipe 30 connected: (<_UnixReadPipeTransport fd=30 polling>, <ReadSubprocessPipeProto fd=2 pipe=<_UnixReadPipeTransport fd=30 polling>>)
[32mINFO    [0m asyncio:base_events.py:1685 execute program '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor': <_UnixSubprocessTransport pid=13729 running stdin=<_UnixWritePipeTransport fd=27 idle bufsize=0> stdout=<_UnixReadPipeTransport fd=28 polling> stderr=<_UnixReadPipeTransport fd=30 polling>>
[32mINFO    [0m asyncio:unix_events.py:514 <_UnixReadPipeTransport fd=30 polling> was closed by peer
[35mDEBUG   [0m asyncio:unix_events.py:1404 process 13729 exited with returncode 1
[32mINFO    [0m asyncio:unix_events.py:514 <_UnixReadPipeTransport fd=28 polling> was closed by peer
[32mINFO    [0m asyncio:unix_events.py:653 <_UnixWritePipeTransport fd=27 idle bufsize=0> was closed by peer
[32mINFO    [0m asyncio:base_subprocess.py:211 <_UnixSubprocessTransport pid=13729 running stdin=<_UnixWritePipeTransport closed fd=27 closed> stdout=<_UnixReadPipeTransport closed fd=28 closed> stderr=<_UnixReadPipeTransport closed fd=30 closed>> exited with return code 1
- generated xml file: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml -
=========================== short test summary info ============================
FAILED test/test_processor_service.py::test_processor_service - test_processo...
[31m============================== [31m[1m1 failed[0m[31m in 0.18s[0m[31m ===============================[0m
